src/Anthropic.Client/SseAggregator.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Anthropic.Client;
+
+/// <summary>
+/// Defines the base for all Aggregators using ServerSideStreaming events.
+/// </summary>
+/// <typeparam name="TMessage"></typeparam>
+/// <typeparam name="TResult"></typeparam>
+public abstract class SseAggregator<TMessage, TResult> 
+    where TResult: new()
+{
+    private readonly IAsyncEnumerable<TMessage> _messages;
+
+    /// <summary>
+    /// Initialises a new instance of the <see cref="SseAggregator{TMessage, TResult}"/>.
+    /// </summary>
+    /// <param name="messages">The service attached enumerable for reading messages.</param>
+    public SseAggregator(IAsyncEnumerable<TMessage> messages)
+    {
+        this._messages = messages;
+    }
+
+    private Task<TResult>? _collectionTask;

```

Comment:
What's the reason to have this field? It doesn't seem to be read as far as I can tell